// @imagina/qmenu/_config/backendPages.js
export default {
//Menu List
  menus: {
    permission: 'menu.menus.index',
    activated: true,
    path: '/menus',
    name: 'qmenu.admin.menus',
    page: () => import('@imagina/qmenu/_pages/admin/menus/index'),
    layout: () => import('@imagina/qsite/_layouts/master.vue'),
    title: 'qmenu.sidebar.adminMenus',
    icon: 'fas fa-stream',
    authenticated: true,
  },
  menusShow: {
    permission: 'menu.menus.edit',
    activated: true,
    path: '/menus/menu/:id',
    name: 'qmenu.admin.menus.show',
    page: () => import('@imagina/qmenu/_pages/admin/menus/show'),
    layout: () => import('@imagina/qsite/_layouts/master.vue'),
    title: 'qmenu.sidebar.adminMenus',
    icon: 'fas fa-stream',
    authenticated: true,
  },
  menuItemsCreate: {
    permission: null,
    activated: true,
    path: '/menus/item/:menuId',
    name: 'qmenu.admin.menu.create',
    page: () => import('@imagina/qmenu/_pages/admin/menuitems/form'),
    layout: () => import('@imagina/qsite/_layouts/master.vue'),
    title: 'qmenu.sidebar.adminMenuItems',
    icon: 'fas fa-circle',
    authenticated: true,
  },
  menuItemsUpdate: {
    permission: null,
    activated: true,
    path: '/menus/menu/:menuId/item/:id',
    name: 'qmenu.admin.menu.update',
    page: () => import('@imagina/qmenu/_pages/admin/menuitems/form'),
    layout: () => import('@imagina/qsite/_layouts/master.vue'),
    title: 'qmenu.sidebar.adminMenuItems',
    icon: 'fas fa-circle',
    authenticated: true,
  },
}
