// @imagina/qlocations/_i18n/en-us/layout.js
export default {
  newCountry: 'New Country',
  updateCountry: 'Edit Country',
  newProvince: 'Nuevo State/Province/Department',
  updateProvince: 'Edit State/Province/Department',
  newCity: 'New City',
  updateCity: 'Editar City',
  newPolygon: 'New Polygon',
  updatePolygon: 'Edit Polygon',
  newGeozone: 'New Geo Zone',
  updateGeozone: 'Edit Geo Zone',
  newNeighborhood: 'New Neighborhood',
  updateNeighborhood: 'Edit Neighborhood',
  form:{
    province: 'State/Province/Department',
    country: 'Country',
    city: 'City',
    polygon: 'Polygon',
    neighborhood: 'Neighborhood',
    provinces: 'States/Provinces/Departments',
    countries: 'Countries',
    cities: 'Cities',
    polygons: 'Polygons',
    neighborhoods: 'Neighborhoods',
    fullName: 'Full Name',
    currency: 'Currency',
    currencySymbol: 'Currency Symbol',
    currencyCode: 'Currency Code',
    currencySubUnit: 'Currency sub Unit',
    regionCode: 'Region Code',
    subRegionCode: 'Sub-Region Code',
    callingCode: 'Calling Code',
    countryCode: 'Country Code',
    iso2: 'ISO Code 2',
    iso3: 'ISO Code 3',
    code: 'Code',
    points: 'Points',
    geozone : 'Geo zone'
  }
}
