// @imagina/qcommerce/_i18n/es/layout.js
export default {
  newProduct : 'Nuevo Producto',
  updateProduct : 'Actualizar Producto',
  newCategory : 'Nueva Categoría',
  updateCategory : 'Actualizar Categoría',
  productOptions : 'Opciones de productos',
  newOption : 'Nueva Opción',
  updateOption : 'Actualizar Opción',
  newOptionValue : 'Nuevo valor de Opción',
  updateOptionValue : 'Actualizar valor de Opción',
  newProductOptionValue : 'Nuevo valor de Opción de producto',
  updateProductOptionValue : 'Actualizar valor de Opción de producto',
  newProductDiscount : 'Nuevo Descuento de producto',
  updateProductDiscount : 'Actualizar Descuento producto',
  updatePaymentMethod : 'Actualizar metodo de pago',
  updateShippingMethod : 'Actualizar metodo de envio',
  newOrder : 'Nuevo Orden',
  updateOrder : 'Actualizar Orden',
  newStore : 'Nueva Tienda',
  updateStore : 'Actualizar Tienda',
  newManufacturer : 'Nuevo Fabricante',
  updateManufacturer : 'Actualizar Fabricante',
  newTaxClass : 'Nuevo Tipo de Impuesto',
  updateTaxClass : 'Actualizar Tipo de Impuesto',
  newTaxRate : 'Nueva Tasa de Impuesto',
  updateTaxRate : 'Actualizar Tasa de Impuesto',
  newCoupon: 'Nuevo Cupon',
  updateCoupon: 'Actualizar Cupon',
  newStatus : 'Nuevo Estado',

  orderAccountInformation: 'Información de pedido y cuenta',
  orderStatus: 'Estado del pedido',
  orderDate: 'Fecha de orden',
  placedFromIP: 'Pedido realizado desde la IP',
  accountInformation: 'Información de la cuenta',
  addressInformation: 'Datos del Domicilio',
  billingAddress: 'Dirección de Envio',
  paymentShippingMethod: 'Pago y método de envío',
  paymentInformation: 'Información del pago',
  itemsOrdered: 'Artículos ordenados',
  orderTotal: 'Total del pedido',
  subtotal: 'Total parcial',
  shipping: 'Envío',
  orderHistory: 'Historial de Orden',

  form : {
    availableDate : 'Disponible desde',
    minimumOrder : 'Orden minima',
    orderWeight : 'Peso de la Orden',
    requriedShipping : 'Requiere envío',
    freeShipping : 'Envío gratis',
    subtractFromStock : 'Descontar de inventarío',
    relatedProducts : 'Productos relacionados',
    optionValue : 'Valor de opción',
    parentOptionValue : 'Valor de opción padre',
    productOptionValues : 'Valores de opción de producto',
    showInMenu:'Mostrar en menu',
    apiKey : 'Codigo API',
    apiLogin : 'API Inicio de sesión',
    transactionKey : 'Codigo de transacción',
    clientKey : 'Codigo de cliente',
    clientSecret : 'Codigo Secreto',
    merchantId : 'ID Comerciante',
    mode : 'Modo',
    accountId : 'ID Cuenta',
    allowTestCreditCard : 'Tarjeta de credito de prueba',
    cost : 'Costo',
    minimum : 'Minimo',
    deliveryFee : 'Cuota de envio',
    userId : 'ID Usuario',
    zipCode : 'Codigo postal',
    shippingRates : 'Tarifas de envio',
    machinable : 'Mecanizable',
    accessKey : 'Codigo de Acceso',
    shipperPostalCode : 'Codigo postal de envio',
    shipperStateCode : 'Codigo de Estado de envio',
    shipperCountryCode : 'Codigo De Pais de envio',
    total: 'Total',
    city: 'Ciudad',
    country: 'País',
    shippingMethod: 'Métodos de Envío',
    dimensions: 'Dimensiones',
    link: 'Relaciones',
    content: 'Contenido',
    data: 'Datos',
    taxRate: 'Tasa de impuesto',
    taxBased: 'Basado en',
    priority: 'Prioridad',
    rate: 'Tasa',
    discountType: 'Tipo de Descuento',
    usersGroup: 'Grupo de Usuarios',
    manufacturer: 'Fabricante',
    customerGroup: 'Grupo de Usuario',
    taxClass: 'Tipo de Impuesto',
    
    // Cupones
    code: 'Codigo de Cupon',
    type: 'Tipo',
    category: 'Categoría',
    categories: 'Categorías',
    product: 'Producto',
    customer: 'Cliente',
    discount: 'Descuento',
    typeDiscount: 'Tipo de Descuento',
    logged: 'Requiere inicio de Sesión',
    shipping: 'Envío',
    dateStart: 'Fecha de Inicio',
    dateEnd: 'Fecha Fin',
    quantityTotal: 'Cantidad Total de Cupones',
    quantityTotalCustomer: 'Cantidad Total de Cupones por Cliente',
    status: 'Estado',
    color: 'Color',
  },
  message : {
    warnAddOpt : 'Para agregar opciones, primero debe crear el producto',
    noOptions : 'No hay opciones agregadas',
    noDeleteOption : 'No puede borrar está opción, es padre de {name}',
    selectProduct : 'Seleccione una opción de producto',
    warnAddDiscount : 'Para agregar descuentos, primero debe crear el producto',
  },
  options:{
    yes: 'Sí',
    no: 'No',
    fixedValue: 'Valor fijo',
    fixed: 'Valor fijo',
    percentage: 'Porcentaje',
    taxRateBased:{
      shippingAddress: 'Dirección de Envío',
      paymentAddress: 'Dirección de Pago',
      storeAddress: 'Dirección de la tienda',
    }
  }
}
