// @imagina/qcommerce/_i18n/es/sidebar.js
export default {
  adminGroup : 'Tienda',
  adminProducts : 'Productos',
  adminProductCreate : 'Creación de Productos',
  adminProductEdit : 'Editar Producto',
  adminCategories : 'Categorias de Producto',
  adminOptions : 'Opciones de Producto',
  adminOptionsCreate : 'Nueva Opción de Producto',
  adminOptionsEdit : 'Editar Opción de Producto',
  adminValues : 'Valores de Opción',
  adminValuesCreate : 'Nuevo Valor de Opción',
  adminValuesEdit : 'Editar Valor de Opción',
  adminPaymentMethods : 'Métodos de pago',
  adminShippingMethods : 'Métodos de envío',
  adminOrders : 'Ordenes',
  orderIndex : 'Orden',
  adminCoupons: 'Cupones',
  createCoupons: 'Crear cupon',
  editCoupons: 'Editar cupon',

  adminStores : 'Tiendas',
  adminStoresCreate : 'Nueva Tienda',
  adminStoresEdit : 'Editar Tienda',
  adminManufacturers : 'Fabricantes',
  adminManufacturersCreate : 'Nuevo Fabricante',
  adminManufacturersEdit : 'Editar Fabricante',
  adminTaxClasses : 'Tipos de impuestos',
  adminTaxClassesCreate : 'Nuevo Tipo de Impuesto',
  adminTaxClassesEdit : 'Editar Tipo de Impuesto',
  adminTaxRates : 'Tasas de Impuesto',
  adminTaxRatesCreate : 'Nueva Tasa de Impuesto',
  adminTaxRatesEdit : 'Editar Tasa de Impuesto',
}
