// @imagina/qlocations/_i18n/es/layout.js
export default {
  newCountry: 'Nuevo País',
  updateCountry: 'Editar País',
  newProvince: 'Nuevo Departamento/Provincia/Estado',
  updateProvince: 'Editar Departamento/Provincia/Estado',
  newCity: 'Nueva Ciudad',
  updateCity: 'Editar Ciudad',
  newPolygon: 'Nuevo Polígono',
  updatePolygon: 'Editar Polígono',
  newGeozone: 'Nuevo Geo Zona',
  updateGeozone: 'Editar Geo Zona',
  newNeighborhood: 'Nuevo Barrio',
  updateNeighborhood: 'Editar Barrio',
  form:{
    province: 'Departamento/Provincia/Estado',
    country: 'País',
    city: 'Ciudad',
    polygon: 'Polígono',
    neighborhood: 'Barrio',
    provinces: 'Departamentos/Provincias/Estados',
    countries: 'Países',
    cities: 'Ciudades',
    polygons: 'Polígonos',
    neighborhoods: 'Barrios',
    fullName: 'Nombre Completo',
    currency: 'Moneda',
    currencySymbol: 'Símbolo de moneda',
    currencyCode: 'Cód. de moneda',
    currencySubUnit: 'Submúltiplo de moneda',
    regionCode: 'Cód. Región',
    subRegionCode: 'Cód. Sub-Region',
    callingCode: 'Código de Área (Para llamadas)',
    countryCode: 'Código del País',
    iso2: 'Cód. ISO 2',
    iso3: 'Cód. ISO 3',
    code: 'Código',
    points: 'Puntos',
    geozone : 'Geo zona'
  }
}
