// @imagina/qsite/_i18n/master/en-us/ui.js
export default {
  label: {
    delete: 'Delete',
    cancel: 'Cancel',
    open: 'Open',
    close: 'Close',
    closed: 'Closed',
    set: 'Set',
    ok: 'Ok',
    select: 'Select',
    reset: 'Reset',
    remove: 'Remove',
    update: 'Update',
    create: 'Create',
    search: 'Search',
    filter: 'Filter | Filters',
    refresh: 'Refresh',
    move: 'Move',
    clear: 'Clear',
    loading: 'Loading',
    validating: 'Validating',
    save: 'Save',
    confirm: 'Confirm',
    configuration: 'Configuration | Configurations',
    role: 'Role | Roles',
    department: 'Department | Departments',
    language: 'Language | Languages',
    interface: 'Interface | Interfaces',
    currency: 'Currency | Currencies',
    setting: 'Setting | Settings',
    tag: 'Tag | Tags',
    description: 'Description | Descriptions',
    option: 'Option | Options',
    edit: 'Edit',
    addFile: 'Add File',
    error: 'Error',
    enabled: 'Enable',
    disabled: 'Disable',
    available: 'Available',
    soldOut: 'Sold Out',
    all: 'All | All',
    data: 'Data',
    link: 'Link',
    image: 'Image | Images',
    add: 'Add',
    required: 'Required',
    yes: 'Yes',
    no: 'No',
    permission: 'Permission | Permissions',
    allow: 'Allow',
    deny: 'Deny',
    inherit: 'Inherit',
    day: 'Day | Days',
    monday: 'Monday',
    tuesday: 'Tuesday',
    wednesday: 'Wednesday',
    thursday: 'Thursday',
    friday: 'Friday',
    saturday: 'Saturday',
    sunday: 'Sunday',
    hour: 'Hour | Hours',
    minute: 'Minute | Minutes',
    second: 'Second | Seconds',
    home: 'Home',
    find: 'Find',
    user: 'User',
    impersonate: 'Impersonate',
    address: 'Address | Addresses',
    contacts: 'Contact | Contacts',
    socialNetwork: 'Social Network | Social Networks',
    success: 'Success',
    ready: 'Ready',
    module: 'Module | Modules',
    entity: 'Entity',
    change: 'Change',
    socialNetwork: 'Social Network | Social Networks',
    logo: 'Logo | Logos',
    pending: 'Pending | Pendings',
    notification: 'Notification | Notifications',
    leave: 'Leave',
    action: 'Action | Actions',
    view: 'View | Views',
    grid: 'Grid | Grids',
    table: 'Table',
    expired: 'Expired',
    empty: 'Empty',
    completed: 'Completed | Completed',
    comment: 'Comment | Comments',
    team: 'Team | Teams',
    gallery: 'Gallery | Galleries',
    event: 'Event | Events',
    profile: 'Profile | Profiles',
    place: 'Place | Places',
    welcome: 'Welcome',
    message: 'Message | Mesages',
    chat: 'Chat | Chats',
    pagination: 'Pagination',
    page: 'Page | Pages',
    today: 'Today',
    yesterday: 'Yesterday',
    tomorrow: 'Tomorrow',
    LastNumDays: 'Last {numDays} days',
    currentWeek: 'Current week',
    lastWeek: 'Last week',
    nextWeek: 'Next week',
    currentMonth: 'Current Month',
    lastMonth: 'Last month',
    nextMonth: 'Next month',
    numMonthsAgo: '{numMonths} months ago',
    currentYear: 'Current year',
    lastYear: 'Last year',
    numYearsAgo: '{numYears} years ago',
    lastNumYears: 'Last {numYears} years',
    customRange: 'Custom range',
    back: 'Back',
    time: 'Time | Times',
    send: 'Send',
    product: 'Product | Products',
    information: 'Information',
    theme: 'Theme | Themes',
    color: 'Color | Colors',
    visible: 'Visible | Visibles',
    featured: 'Feautred | Featured',
    share : 'Share',
    unit: 'Unit | Units',
    showMore : 'Show more',
    date: 'Date | Dates',
    email: 'Email | Emails',
    phone: 'Phone | Phones',
    birthday: 'Birthday',
    text : 'Text | Texts'
  },
  message: {
    notFound: 'Sorry, nothing here',
    goToHome: 'Go to home',
    deleteRecord: 'Are you sure, you want to delete this record?',
    formInvalid: 'There are missing fields to complete, check the form',
    fieldRequired: 'This field is required',
    fieldEmail: 'Must have email format',
    fieldCheckPassword: 'Does not match the password',
    fieldMinLeng: 'Must contain at least {num} characters',
    fieldMaxLeng: 'Please use maximum {num} characters',
    updatePassword: 'Update password',
    recordCreated: 'Record created',
    recordUpdated: 'Record updated',
    recordDeleted: 'Record deleted',
    recordNoCreated: 'Record NOT created',
    recordNoUpdated: 'Record NOT updated',
    recordNoDeleted: 'Record NOT deleted',
    errorRequest: 'Error requesting data',
    saveAndReturn: 'Save and Return',
    saveAndEdit: 'Save and continue editing',
    saveAndCreate: 'Save and create new',
    searchByIdOrName: 'Search by ID or Name',
    searchNotFound: 'Search not found',
    uploadImage: 'Upload image',
    changeImage: 'Change image',
    editImage: 'Edit image',
    deleteImage: 'Delete image',
    captcha: 'This site is protected by reCAPTCHA and the Google ' +
      '<a href="https://policies.google.com/privacy" target="_blank">Privacy Policy</a> and ' +
      '<a href="https://policies.google.com/terms" target="_blank">Terms of Service</a> apply.',
    requiredCaptcha: 'Captcha is required',
    deniedPermissions: "You don't have the necessary permissions",
    requestSent: 'Request Sent',
    titleAlreadyExist: 'The title already exists',
    fieldAlreadyExist: 'The field already exists',
    pageDataWithFilter: 'The information on this page has filters',
    copiedToClipboard : 'Copied to clipboard',
    preview : 'Preview',
    showMore : 'Show more',
    request : 'Request | Request',
    accept : 'Accept',
    reject : 'Reject',
    rejected : 'Rejected',
    changeTo : 'Change to {text}'
  },
  form: {
    id: 'ID',
    title: 'Title | Title',
    caption: 'Caption | Captions',
    firstName: 'First Name | First Name',
    lastName: 'Last Name | Last Name',
    password: 'Pasword | Paswords',
    checkPassword: 'Check Pasword | Check Paswords',
    name: 'Name | Names',
    email: 'Email | Emails',
    slug: 'Slug',
    summary: 'Summary | Summaries',
    description: 'Description Descriptions',
    category: 'Category | Categories',
    parent: 'Parent | Parents',
    status: 'status | Statuses',
    image: 'Image | Images',
    video: 'Video | Videos',
    firstImage: 'First Image',
    secondaryImage: 'Secondary Image',
    gallery: 'Gallery | Galleries',
    tag: 'Tag | Tag',
    publish: 'Published',
    noPublish: 'Not Published',
    draft: 'draft',
    pending: 'Pending',
    metaTitle: 'Meta Title',
    metaDescription: 'Meta Description',
    sku: 'Sku',
    stock: 'stock',
    price: 'Price | Prices',
    quantity: 'Quantity | Quanitities',
    weight: 'Weight',
    length: 'Length',
    height: 'Height',
    width: 'Width',
    point: 'Point | Points',
    reference: 'Reference',
    option: 'Option | Options',
    type: 'Type | Types',
    required: 'Required',
    createdAt: 'Created At',
    updatedAt: 'Updated At',
    actions: 'Actions',
    sort: 'Sort',
    prefix: 'Prefix',
    currentNumber: 'Current number',
    firstNumber: 'First number',
    lastNumber: 'Last number',
    resolution: 'Resolution',
    autoincrement: 'autoincrement',
    default: 'Default',
    percentage: 'Porcentage',
    address: 'Address | Addresses',
    phone: 'Phone | Phones',
    schedule: 'Schedule | Schedules',
    menu: 'Menu | Menus',
    author: 'Author | Authors',
    icon: 'Icon | Icons',
    masterRecord: 'Master record',
    birthday: 'Birthday',
    identification: 'Identification',
    city: 'City | cities',
    state: 'State | State',
    province: 'Province | Provinces',
    zipCode: 'Zip Code',
    socialNetwork: 'Social Network | Social Networks',
    startDate: 'Start Date',
    endDate: 'End Date',
    code: 'Code | Codes',
    increase: 'Increase',
    date: 'Date | Dates',
    hour: 'Hour | Hours',
    dueDate: 'Due date',
    tax: 'Tax | Taxes',
    orderBy: 'Order by',
    orderWay: 'Order way',
    assignedTo: 'Assigned to',
    comment: 'Comentario | Comentarios',
    biography: 'Biography | Biographies',
    page: 'Page | Pages',
    perPage: 'Per page',
    userName: 'User Name',
    role: 'Role | Roles',
    department: 'Department | Departments',
    globalSearch: 'Global Search',
    urlMap: 'Map URL',
    primaryColor: 'Primary color',
    secondaryColor: 'Secondary color',
    tertiaryColor: 'Tertiary color',
    priceOffer : 'Offert price',
    onOffer : 'On offert',
    createdBy : 'Created by',
    systemName : 'System name'
  },
  configList: {
    loggedAs: 'logged as',
    fullScreen: 'show in full screen',
    signOut: 'sign out',
    leaveImpersonating: 'Leave impersonating'
  },
}
