// @imagina/qsite/_i18n/master/es/ui.js
export default {
  label: {
    delete: 'Eliminar',
    cancel: 'Cancelar',
    open: 'Abierto',
    close: 'Cerrar',
    closed: 'Cerrado',
    set: 'Poner',
    ok: 'Ok',
    select: 'Seleccionar',
    reset: 'Resetear',
    remove: 'Remover',
    update: 'Actualizar',
    create: 'Crear',
    search: 'Buscar',
    filter: 'Filtro | Filtros',
    refresh: 'Refrescar',
    move: 'Mover',
    clear: 'Limpiar',
    loading: 'Cargando',
    validating: 'Validando',
    save: 'Guardar',
    confirm: 'Confirmar',
    configuration: 'Configuración | Configuraciones',
    role: 'Rol | Roles',
    department: 'Departamento | Departamentos',
    language: 'Idioma | Idiomas',
    interface: 'Interfaz | Interfaces',
    currency: 'Moneda | Monedas',
    setting: 'Configuración | Configuraciones',
    tag: 'Etiqueta | Etiquetas',
    description: 'Descripción | Descripciones',
    option: 'Opción | Opciones',
    edit: 'Editar',
    addFile: 'Examinar',
    error: 'Error',
    enabled: 'Activo',
    disabled: 'Inactivo',
    available: 'Disponible',
    soldOut: 'Agotado',
    all: 'Todo | Todos',
    data: 'Dato',
    link: 'Enlazar',
    image: 'Imagen | Imagenes',
    add: 'Agregar',
    required: 'Requerido',
    yes: 'Si',
    no: 'No',
    permission: 'Permiso | Permisos',
    allow: 'Permitir',
    deny: 'Negar',
    inherit: 'Heredar',
    day: 'Día | Dias',
    monday: 'Lunes',
    tuesday: 'Martes',
    wednesday: 'Miercoles',
    thursday: 'Jueves',
    friday: 'Viernes',
    saturday: 'Sabado',
    sunday: 'Domingo',
    hour: 'Hora | Horas',
    minute: 'Minuto | Minutos',
    second: 'Segundo | Segundos',
    home: 'Inicio',
    find: 'Buscar',
    user: 'Usuario',
    impersonate: 'Suplantar',
    address: 'Dirección | Direcciones',
    contact: 'Contacto | Contactos',
    socialNetwork: 'Red Social | Redes Sociales',
    success: 'Realizado',
    ready: 'Listo',
    module: 'Modulo',
    entity: 'Entidad',
    change: 'Cambiar',
    socialNetwork: 'Red social | Redes sociales',
    logo: 'Logo | Logos',
    pending: 'Pendiente | Pendientes',
    notification: 'Notificación | Notificaciones',
    leave: 'Abandonar',
    action: 'Acción | Acciones',
    view: 'Vista | Vistas',
    grid: 'Cuadricula | Cuadriculas',
    table: 'Tabla',
    expired: 'Caducado',
    empty: 'Vacío',
    completed: 'Completado | Completados',
    comment: 'Comentario | Comentarios',
    team: 'Equipo | Equipos',
    gallery: 'Galería | Galerías',
    event: 'Evento | Eventos',
    profile: 'Perfil | Perfiles',
    place: 'Lugar | Lugares',
    welcome: 'Bienvenido | Bienvenidos',
    message: 'Mensaje | Mensajes',
    chat: 'Chat | Chats',
    pagination: 'Paginación',
    page: 'Pagína | Paginas',
    today: 'Hoy',
    yesterday: 'Ayer',
    tomorrow: 'Mañana',
    LastNumDays: 'Ultimos {numDays} dias',
    currentWeek: 'Semana actual',
    lastWeek: 'Ultima semana',
    nextWeek: 'Siguiente semana',
    currentMonth: 'Mes actual',
    lastMonth: 'ultimo mes',
    nextMonth: 'Siguiente mes',
    numMonthsAgo: 'Hace {numMonths} meses',
    currentYear: 'Año actual',
    lastYear: 'Ultimo año',
    numYearsAgo: 'Hace {numYears} años',
    lastNumYears: 'Ultimos {numYears} años',
    customRange: 'Rango personalizado',
    back: 'Volver',
    time: 'Tiempo | Tiempos',
    send: 'Enviar',
    product: 'Producto | Productos',
    information: 'Información',
    theme: 'Tema | Temas',
    color: 'Color | Colores',
    visible: 'Visible | Visibles',
    featured: 'Destacado | Destacados',
    share: 'Compartir',
    unit: 'Unidad | Unidades',
    showMore: 'Ver más',
    date: 'Fecha | Fechas',
    email: 'Correo | Correos',
    phone: 'Teléfono | Teléfonos',
    birthday: 'Cumpleaños',
    request: 'Solicitud | Solicitudes',
    accept: 'Aceptar',
    reject: 'Rechazar',
    rejected: 'Rechazado',
    text: 'Texto | Textos'
  },
  message: {
    notFound: 'Lo sentimos, no hay nada aquí',
    goToHome: 'Volver al inicio',
    deleteRecord: '¿Estás seguro que quieres eliminar este registro?',
    formInvalid: 'Faltan campos por completar, verifique el formulario',
    fieldRequired: 'Este campo es requerido',
    fieldEmail: 'Debe tener formato de email',
    fieldCheckPassword: 'No coincide con la contraseña',
    fieldMinLeng: 'Debe contener al menos {num} caracteres',
    fieldMaxLeng: 'Debe contener maximo {num} caracteres',
    updatePassword: 'Actualizar contraseña',
    recordCreated: 'Registro creado',
    recordUpdated: 'Registro actualizado',
    recordDeleted: 'Registro eliminado',
    recordNoCreated: 'Registro NO creado',
    recordNoUpdated: 'Registro NO actualizado',
    recordNoDeleted: 'Registro NO eliminado',
    errorRequest: 'Error al solicitar los datos',
    saveAndReturn: 'Guardar y regresar',
    saveAndEdit: 'Guardar y continuar editando',
    saveAndCreate: 'Guardar y crear uno nuevo',
    searchByIdOrName: 'Buscar por ID o nombre',
    searchNotFound: 'Busqueda sin resultados',
    uploadImage: 'Subir imagen',
    changeImage: 'Cambiar imagen',
    editImage: 'Editar imagen',
    deleteImage: 'Borrar imagen',
    captcha: 'Este sitio está protegido por reCaptcha y se aplican las ' +
      '<a href="https://policies.google.com/privacy" target="_blank">Politicas de Privacidad</a> y los' +
      '<a href="https://policies.google.com/terms" target="_blank">Terminos de servicios</a> de Google.',
    requiredCaptcha: 'Captcha es requerida',
    deniedPermissions: "No tienes los permisos necesarios para realizar esta acción",
    requestSent: 'Solicitud enviada',
    titleAlreadyExist: 'El titulo ya existe',
    fieldAlreadyExist: 'El campo ya existe',
    pageDataWithFilter: 'La información en esta página tiene filtros',
    copiedToClipboard: 'Copiado en el portapapeles',
    preview: 'Vista previa',
    showMore: 'Ver más',
    changeTo: 'Cambiar a {text}'
  },
  form: {
    id: 'ID',
    title: 'Título | Titulos',
    caption: 'Subtítulo | Subtítulos',
    firstName: 'Nombre | Nombres',
    lastName: 'Apellido | Apellidos',
    password: 'Contraseña | Contraseñas',
    checkPassword: 'Confirmar Contraseña | Confirmar Contraseñas',
    name: 'Nombre | Nombres',
    email: 'Email | Emails',
    slug: 'Slug',
    summary: 'Resumen | Resumenes',
    description: 'Descripción | Descripciones',
    category: 'Categoría | Categorías',
    parent: 'Padre | Padres',
    status: 'Estado | Estados',
    image: 'Imagen | Imagenes',
    video: 'Vídeo | Vídeos',
    firstImage: 'Imagen Principal',
    secondaryImage: 'Imagen Secundaría',
    gallery: 'Galería | Galerías',
    tag: 'Etiqueta | Etiquetas',
    publish: 'Publicado',
    noPublish: 'No Publicado',
    draft: 'Borrador',
    pending: 'Pendiente',
    metaTitle: 'Meta Titulo',
    metaDescription: 'Meta Descripción',
    sku: 'Sku',
    stock: 'Existencía',
    price: 'Precio | Precios',
    quantity: 'Cantidad | Cantidades',
    weight: 'Peso',
    length: 'Largo',
    height: 'Alto',
    width: 'Ancho',
    point: 'Punto | Puntos',
    reference: 'Referencía',
    option: 'Opción | Opciones',
    type: 'Tipo | Tipos',
    required: 'Requerido',
    createdAt: 'Creado en',
    updatedAt: 'Actualizado en',
    actions: 'Acciones',
    sort: 'Ordenar',
    prefix: 'Prefijo',
    currentNumber: 'Numero actual',
    firstNumber: 'Numero inicial',
    lastNumber: 'Numero final',
    resolution: 'Resolución',
    autoincrement: 'Autoincrementar',
    default: 'Por defecto',
    percentage: 'Porcentaje',
    address: 'Dirección | Direcciones',
    phone: 'Teléfono | Teléfonos',
    schedule: 'Horario | Horarios',
    menu: 'Menu | Menus',
    author: 'Autor | Autores',
    icon: 'Icono | Iconos',
    masterRecord: 'Registro maestro',
    birthday: 'Cumpleaños',
    identification: 'Identificación',
    city: 'Ciudad | Ciudades',
    state: 'Departamento | Departamentos',
    province: 'Provincia | Provincias',
    zipCode: 'Codigo Postal',
    socialNetwork: 'Red Social | Redes Sociales',
    startDate: 'Fecha Inicio',
    endDate: 'Fecha Fin',
    code: 'Codigo | Codigos',
    increase: 'Incrementar',
    date: 'Fecha | Fechas',
    hour: 'Hora | Horas',
    dueDate: 'Fecha de vencimiento',
    tax: 'Impuesto | Impuestos',
    orderBy: 'Ordenar por',
    orderWay: 'Ordenar de forma',
    assignedTo: 'Asignado a',
    comment: 'Comentario | Comentarios',
    biography: 'Biografía | Biografias',
    page: 'Pagína | Paginas',
    perPage: 'Por pagína',
    userName: 'Nombre de usuario',
    role: 'Rol | Roles',
    department: 'Departamento | Departamentos',
    globalSearch: 'Búsqueda Global',
    urlMap: 'URL del mapa',
    primaryColor: 'Color primario',
    secondaryColor: 'Color secondario',
    tertiaryColor: 'Color terciario',
    priceOffer: 'Precio de oferta',
    onOffer: 'En oferta',
    createdBy: 'Creado por',
    systemName: 'Nombre de sistema',
  },
  configList: {
    loggedAs: 'Registrado como',
    fullScreen: 'Pantalla completa',
    signOut: 'Cerrar sesión',
    leaveImpersonating: 'Dejar de suplantar'
  },
}
