// @imagina/qcommerce/_i18n/en-us/sidebar.js
export default {
  adminGroup : 'Commerce',
  adminProducts : 'Product List',
  adminProductCreate : 'Product Create',
  adminProductEdit : 'Product Edit',
  adminCategories : 'Product Categories',
  adminOptions : 'Product Options',
  adminOptionsCreate : 'Create Product Option',
  adminOptionsEdit : 'Edit Product Optio',
  adminValues : 'Options Values',
  adminValuesCreate : 'Create Option Value',
  adminValuesEdit : 'Edit Option Value',
  adminPaymentMethods : 'Payment methods',
  adminShippingMethods : 'Shipping methods',
  adminOrders : 'Orders',
  orderIndex : 'Order',
  adminCoupons: 'Coupons',
  createCoupons: 'Create cupon',
  editCoupons: 'Edit cupon',

  adminStores : 'Stores',
  adminStoresCreate : 'Create Store',
  adminStoresEdit : 'Edit Store',
  adminManufacturers : 'Manufacturers',
  adminManufacturersCreate : 'Create Manufacturer',
  adminManufacturersEdit : 'Edit Manufacturers',
  adminTaxClasses : 'Tax Classes',
  adminTaxClassesCreate : 'New Tax Class',
  adminTaxClassesEdit : 'Edit Tax Class',
  adminTaxRates : 'Tax Rates',
  adminTaxRatesCreate : 'New Tax Rate',
  adminTaxRatesEdit : 'Edit Tax Rate',
}
